### DISCUSSION TITLE
Want to see your feature requests on our roadmap?

### DISCUSSION THREAD

**Discussion by joebudi** [Budibase/budibase#5518] [Polls]
We're using this poll to help figure out what features we should build next and potentially source beta testers.

**Comment by poirazis** [comment]
Hey @joebudi 

I would like to suggest another one! Reusable Element Blocks 
That would be group of elements you can save and reuse, similar to the Reusable Blocks of Wordpress Gutenberg.
https://wordpress.com/support/wordpress-editor/blocks/reusable-block/


Could be functionality of "Custom Components" i guess.

**Reply by joebudi** [reply]
Great suggestion and something we have thought about a lot recently!

**Reply by cantonsen** [reply]
I agree. This would be a very handy feature!

**Comment by zimmra** [comment]
More granular control over budibase defined relationships of SQL tables - when dealing with large tables relationships can start bogging things down due to it pulling all of the related table/rows's info. 

Would be much more efficient if specific columns could be excluded.

**Reply by mjashanks** [reply]
I agree with this and have thought a lot about this. I'd love to see options like this in the data provider. Theres a bunch of things that we could do with the data provider IMO
- Choose columns
- Choose multiple joins, and select columns from joined tables
- UNIONs
- Allow a dataprovider to fetch multiple data sources at once.

I think this would be great, but I can't see us doing this soon, as it's achievable with custom queries - so it's hard to prioritise over other things. Someday though.

**Comment by Fabian1410** [comment]
Email Automation for Receiving Emails with Attachments :D

**Comment by aderici** [comment]
When you hit a limitation with backend you can simply spin up a low-code backend server and use the http call / rest api functions to solve your problem.

I feel there is more need for a beautiful looking easy to use frontend components. And those would potentially come from the community so maybe people develop custom components and share them as gists and as an admin i can import them?

There is great potential in this tool. Better adopt a relaxed and modifiable style so that people can find workarounds than be strict and late with development (and people hitting the limitations would have to go somewhere else)

**Comment by JUVOJustin** [comment]
I feel like this is a huge internal thing to "fix": https://github.com/Budibase/budibase/issues/4789 .Feels kind of wrong to me that I was made aware of budibase as a tool to build portals and backends, but this is still an issue.

**Comment by kshep92** [comment]
I dabbled with Retool and Appsmith recently and I really like how they give code completion assistance and live evaluations of data binding expressions in their input fields. I'd love to have this in Budibase...

![image](https://user-images.githubusercontent.com/3780572/169742092-586ff4bf-ea4a-486b-a4ef-00dcbe43be6e.png)
![image](https://user-images.githubusercontent.com/3780572/169742303-04d4110d-8070-41bd-855a-7a2e287a1669.png)

**Comment by Shadowfita** [comment]
Custom components would be awesome. Especially if you could construct a component hierarchy and then "group" them together and have them be exportable / importable. This opens the door to lots of powerful components that Budibase doesn't necessarily have to support out-of-the-box.

**Comment by nishantnaagnihotri** [comment]
Automatic Default app access for all users.
#5617 #3559 #2605

**Reply by joebudi** [reply]
User groups are coming in the next few weeks. 👍

**Comment by andresps-dev** [comment]
Hi, I would love to help with the spanish translation! I'm launching Budibase as internal tool on my company and our employees would be very grateful if they can have the UI messages and menus on their language. Just tell me and I'll start, and thanks for your awesome work!

**Reply by fabianlaule** [reply]
Same here for the German translation. If you need help, let me know.

**Reply by airbenich** [reply]
Same here!

**Reply by nunopoco** [reply]
Portuguese translation. 🙋

**Reply by dfrost999** [reply]
Generally i18n + l10n FTW

**Comment by ghost** [comment]
If/else statements in automations!

**Comment by jaz000** [comment]
Export all Attachments (Column or whole table).

More customization of tables (font, size appearance, image handling).

Copying / export / import of design components (screens, selected component hierarchy).

**Comment by acseven** [comment]
After looking multiple issues here on github and from my own experience with Budibase, search features are very much underperformant given the platform's current state. 

When end users think about search, they don't care for letter case, if there are accented characters or not, if a word is in the beginning or the end of a field, or even if the field being looked up is only in a specific column; the user expects the search to output the logic values and the platform is too limited for general use. 

That's my suggestion for focus: heavy search / filtering improvements!

**Comment by agruenberg** [comment]
Custom date format for the date picker component.

**Reply by nunopoco** [reply]
I guess this would fit under a localization feature, as languages, dates, decimal numbers, etc can be represented differently around the world.

**Reply by dfrost999** [reply]
I think this would be a useful/necessary addition!

**Comment by JayP718** [comment]
Maybe not so much of a high priority but modifying the icon component to use either Spectrum or Font Awesome fonts. There's probably a good reason but I'm not sure on why the icon picker uses font awesome vs most other things within Budibase using spectrum.

**Comment by Alisa-Modeste** [comment]
(Semi) Single page app. When I go from one detail page to another (/characters/:id pages for me), just have the data change instead of reloading the components

**Comment by kasgithub8** [comment]
A "does not contain" filter.
Image via https://github.com/Budibase/budibase/discussions/8352#discussioncomment-3936395:
<img width="576" alt="197263404-1140eae5-ea19-41fd-8129-30ccaf481b2f" src="https://user-images.githubusercontent.com/95284278/197456187-4f121f54-dda7-4506-a15e-f936ba0ace54.png">

**Comment by sql4bucks** [comment]
Custom, reusable components constructed with the UI and reusable js functions/snippets would be my top two on the wish list. I am evaluating Budibase to decide on a no/low code platform for our company and the amount of redundant code in just one app I'm building is really hard to manage. 

Closely following would be integration with Git in order to develop and merge branches and make this more team friendly. I need more granular control than "reverting from production".

**Reply by AlejandroFNadal** [reply]
I am doing full app backups to be able to revert, a git integration would be great!

**Comment by rs911** [comment]
I really like retool's development process but it doesn't have good front-end display features. Some ideas you can implement that are available in retool (and other app builders):

1. Add any component to the page - budibase does not allow working on form components outside of a form. And so the amount of UI components available is very little. Which is severely limiting in how you can design things.

2. Allow updating components on the same page based on an action performed, like a button click or select row etc.

3. Data access for components is somewhat complicated and not intuitive. Allow any component to access any data source. If I could run any query from any component, it will make things so much easier. Learning, reading and writing SQL is much easier than full on javascript. Which is why I think it will make development a lot easier.

4. And of course code auto completion is a big help. Especially for non developers.

Maybe some of this functionality already exists but I am having a hard time understanding how everything works in budibase. It was very easy to pick up other tools and understand their limitations. I feel like budibase isn't actually limited in capabilities, but just designed for a specific type of workflow. Which limits how it can be used. Making some of these changes will make it the best low code tool.
